src/extension/CMakeLists.txt

```diff
@@ -5,54 +5,201 @@
 # libraries are not included and need to be separately downloaded
 # (for example from our official Thrive git repository)
 
-add_library(thrive_extension SHARED
-  "${PROJECT_BINARY_DIR}/Include.h"
-  core/RegisterThriveTypes.cpp core/RegisterThriveTypes.hpp
-  core/ThriveConfig.cpp core/ThriveConfig.hpp
-  interop/ExtensionInterop.cpp interop/ExtensionInterop.h
-  nodes/DebugDrawer.cpp nodes/DebugDrawer.hpp
-  core/GodotJoltConversions.hpp
+add_library(thrive_extension SHARED)
+
+# Collect all source files recursively
+file(GLOB_RECURSE EXTENSION_SOURCES 
```

Comment:
Again, I really dislike GLOB because it doesn't automatically re-run cmake when new files are added.

Reply:
if you have another way to do it recursively i'm all ears pal, i added GLOB because it seemed the easier to implement and the one that was pretty easy on the CPUs plus it allows anyone to add anything they want in the folder and just start working on it without editing addittional code other than the interops

Reply:
The only good way is to manually list each file. That way cmake is forced to re-run when a new file is defined and everything works fine. With GLOB you have to *manually* re-run cmake whenever you add a file so it is a major pain and you cannot easily see what all gets compiled in. So I'm totally against the use of GLOB at all for defining cmake files to compile.